lib/spy.js

```diff
@@ -91,59 +251,76 @@ module.exports = function (chai, _) {
   }
```

Comment:
Maybe not as part of this PR, but I'd like to remove the reset method from the spy and push it up into our API - or possibly even get rid of `reset` entirely now.